### ISSUE TITLE
[Feature] Could templates support magical variables like: path datetime ...

### ISSUE THREAD

**Issue by chenxxzhe** [yuanhjty/code-template-tool#13]
just like the variables supported in the Templates Path  `{home}` `{workspace}` and so on

**Comment by yuanhjty** [comment]
I have no idea which magical variables are needed. Could you give some detailed suggestions?

**Comment by SteinTheRuler** [comment]
Variations of current Date is obvious. Author, Author Email, Author Link, Filename (name of the current file with ___name___ applied), Directory (destination). Just off the top of my head

**Comment by yuanhjty** [comment]
Actually, I think it's not very meaningful to have magical variables.  Now days we have version control tools which are powerful enough to record various informations related to our codes.

**Comment by hairlesshobo** [comment]
I agree with @SteinTheRuler with regards to at least the Directory (ideally relative to the workspace root) where the template was generated. This could be very beneficial in the output templates.

**Comment by SteinTheRuler** [comment]
I just think somebody would use this and would be a nice little feature

**Comment by casiso** [comment]
I certainly would like to have at least author, filename, creation date and email as I use them in doxygen directives  (@author etc.) for my documentation. author & email could be predefined in a settings directive...
